buildSrc/src/test/java/org/elasticsearch/gradle/test/rest/transform/TransformTests.java

```diff
@@ -0,0 +1,271 @@
+        transformedTests.stream().filter(node -> node.get("teardown") != null).forEach(this::assertTeardown);
+    }
+
+    protected ObjectNode validateSkipNodesExist(List<ObjectNode> tests) {
+        List<ObjectNode> skipNodes = tests.stream()
+            .filter(node -> node.get("setup") != null)
+            .filter(node -> getSkipNode((ArrayNode) node.get("setup")) != null)
+            .map(node -> getSkipNode((ArrayNode) node.get("setup")))
+            .collect(Collectors.toList());
+        assertThat(skipNodes.size(), CoreMatchers.equalTo(1));
+        return skipNodes.get(0);
+    }
+
+    protected void validateSkipNodesDoesNotExist(List<ObjectNode> tests) {
+        List<ObjectNode> skipNodes = tests.stream()
+            .filter(node -> node.get("setup") != null)
+            .filter(node -> getSkipNode((ArrayNode) node.get("setup")) != null)
+            .map(node -> getSkipNode((ArrayNode) node.get("setup")))
+            .collect(Collectors.toList());
+        assertThat(skipNodes.size(), CoreMatchers.equalTo(0));
+    }
+
+    protected void validatePreExistingFeatureExist(List<ObjectNode> tests) {
+        assertThat(validateSkipNodesExist(tests).get("features"), CoreMatchers.notNullValue());
+    }
+
+    protected void validateSetupDoesNotExist(List<ObjectNode> tests) {
+        assertThat(tests.stream().filter(node -> node.get("setup") != null).count(), CoreMatchers.equalTo(0L));
+    }
+
+    protected void validateFeatureNameExists(List<ObjectNode> tests, String featureName) {
+        ObjectNode skipNode = validateSkipNodesExist(tests);
+        JsonNode featureValues = skipNode.get("features");
+        assertNotNull(featureValues);
+
+        List<String> features = new ArrayList<>(1);
+        if (featureValues.isArray()) {
+            Iterator<JsonNode> featuresIt = featureValues.elements();
+            while (featuresIt.hasNext()) {
+                JsonNode feature = featuresIt.next();
+                features.add(feature.asText());
+            }
+        } else if (featureValues.isTextual()) {
+            features.add(featureValues.asText());
+        }
+
+        assertThat(features, IsCollectionContaining.hasItem(featureName));
+    }
+
+    protected void validateSetupExist(List<ObjectNode> tests) {
+        assertThat(tests.stream().filter(node -> node.get("setup") != null).count(), CoreMatchers.equalTo(1L));
+    }
+
+    protected List<ObjectNode> transformTests(List<ObjectNode> tests) {
+        return transformTests(tests, getTransformations());
+    }
+
+    protected List<ObjectNode> transformTests(List<ObjectNode> tests, List<RestTestTransform<?>> transforms) {
+        List<ObjectNode> t = transformer.transformRestTests(new LinkedList<>(tests), transforms);
+        if (getKnownFeatures() != null) {
+            getKnownFeatures().forEach(name -> { validateFeatureNameExists(t, name); });
+        }
+        return t;
+    }
+
+    protected List<String> getKnownFeatures() {
```

Comment:
would return empty list help with other usages around forEach (the one above for example)?

Reply:
yeah... that is probably better ... updated [`857b2c6` (#69007)](https://github.com/elastic/elasticsearch/pull/69007/commits/857b2c60f679391b8b95f0979d670d85d8349cde)